src/main/java/com/okta/tools/BrowserAuthentication.java

```diff
@@ -0,0 +1,87 @@
+package com.okta.tools;
+
+import com.okta.tools.helpers.CookieHelper;
+import javafx.application.Application;
+import javafx.concurrent.Worker.State;
+import javafx.scene.Group;
+import javafx.scene.Scene;
+import javafx.scene.control.ScrollPane;
+import javafx.scene.web.WebEngine;
+import javafx.scene.web.WebView;
+import javafx.stage.Stage;
+import org.apache.commons.lang.StringUtils;
+
+import java.io.*;
+import java.net.CookieHandler;
+import java.net.URI;
+import java.util.*;
+import java.util.concurrent.CountDownLatch;
+
+public class BrowserAuthentication extends Application {
+    // FIXME: this is awful
+    private static final CountDownLatch LATCH = new CountDownLatch(1);
+    // FIXME: this is worse
+    private static OktaAwsCliEnvironment ENVIRONMENT;
```

Comment:
👎 seriously. This is really nasty and I want a better solution. I'm trying to learn JavaFX, so I can handle this a lot more cleanly with instance-based variables rather than globals.